runtime/lua/vim/lsp/buf.lua

```diff
@@ -448,8 +448,9 @@ function M.document_symbol(opts)
 end
 
 --- @param call_hierarchy_items lsp.CallHierarchyItem[]?
+--- @return lsp.CallHierarchyItem?
 local function pick_call_hierarchy_item(call_hierarchy_items)
-  if not call_hierarchy_items then
+  if not call_hierarchy_items or vim.tbl_isempty(call_hierarchy_items) then
```

Comment:
I'm unsure of whether we should have this check in this function or in the invocation place instead:
https://github.com/neovim/neovim/blob/512d63df5ee3f10f42afe009a9d32dc0113c32d4/runtime/lua/vim/lsp/buf.lua#L480-L482

Since the message in the snippet above refers to the case where the server didn't resolve any items, it makes sense to handle the empty case there IMO.

Reply:
Although I'm unsure of what `nil` versus an empty array means here, and whether we should treat them differently...

Reply:
Okay the protocol doesn't make a distinction between empty and a nil response, and [VS Code treats them the same too](https://github.com/microsoft/vscode/blob/94cd4ed34ca1f718615c0befb1b31181db42575e/src/vs/workbench/api/browser/mainThreadLanguageFeatures.ts#L845), so I do think we should move the check to the linked `if`.

Reply:
Yeah I agree, I make the change